swift/ql/test/query-tests/Security/CWE-321/test.swift

```diff
@@ -0,0 +1,169 @@
+
+// --- stubs ---
+
+// These stubs roughly follows the same structure as classes from CryptoSwift
+class AES
+{
+	init(key: Array<UInt8>, blockMode: BlockMode, padding: Padding) { }
+	init(key: Array<UInt8>, blockMode: BlockMode) { }
+	init(key: String, iv: String) { }
+	init(key: String, iv: String, padding: Padding) { }
+}
+
+class Blowfish
+{
+	init(key: Array<UInt8>, blockMode: BlockMode, padding: Padding) { }
+	init(key: Array<UInt8>, blockMode: BlockMode) { }
+	init(key: String, iv: String) { }
+	init(key: String, iv: String, padding: Padding) { }
+}
+
+class HMAC
+{
+	init(key: Array<UInt8>) { }
+	init(key: Array<UInt8>, variant: Variant) { }
+	init(key: String) { }
+	init(key: String, variant: Variant) { }
+}
+
+class ChaCha20
+{
+	init(key: Array<UInt8>, iv: Array<UInt8>) { }
+	init(key: String, iv: String) { }
+}
+
+class CBCMAC
+{
+	init(key: Array<UInt8>) { }
+}
+
+class CMAC
+{
+	init(key: Array<UInt8>) { }
+}
+
+class Poly1305
+{
+	init(key: Array<UInt8>) { }
+}
+
+class Rabbit
+{
+	init(key: Array<UInt8>) { }
+	init(key: String) { }
+	init(key: Array<UInt8>, iv: Array<UInt8>) { }
+	init(key: String, iv: String) { }
+}
+
+enum Variant {
+	case md5, sha1, sha2, sha3
+}
+
+protocol BlockMode { }
+
+struct CBC: BlockMode { 
+	init() { }
+}
+
+protocol PaddingProtocol { }
+
+enum Padding: PaddingProtocol {
+  case noPadding, zeroPadding, pkcs7, pkcs5, eme_pkcs1v15, emsa_pkcs1v15, iso78164, iso10126
+}
+
+// Helper functions
+func getConstantString() -> String {
+  "this string is constant"
+}
+
+func getConstantArray() -> Array<UInt8> {
+	[UInt8](getConstantString().utf8)
+}
```

Comment:
`getConstantArray` is currently unused in the test.

Reply:
Right. I was planning on using this function for a corner case that deals with constant propagation. It's a true negative though, so I'd have to properly document it in the test.

Reply:
Its fine to have test cases we don't catch or don't yet catch.  Just mark them with `// [NOT DETECTED]` or similar.